apps/admin-x-activitypub/src/components/Profile.tsx

```diff
@@ -42,9 +44,49 @@ function useFollowingCountForUser(handle: string) {
     });
 }
 
+function useFollowingForUser(handle: string) {
+    const site = useBrowseSite();
+    const siteData = site.data?.site;
+    const siteUrl = siteData?.url ?? window.location.origin;
+    const api = new ActivityPubAPI(
+        new URL(siteUrl),
+        new URL('/ghost/api/admin/identities/', window.location.origin),
+        handle
+    );
+    return useQuery({
+        queryKey: [`following:${handle}`],
+        async queryFn() {
+            const followingData = await api.getFollowing();
+            // The data is already in the correct format, no need for additional processing
+            return followingData;
+        }
+    });
+}
+
+function useFollowersForUser(handle: string) {
+    const site = useBrowseSite();
+    const siteData = site.data?.site;
+    const siteUrl = siteData?.url ?? window.location.origin;
+    const api = new ActivityPubAPI(
+        new URL(siteUrl),
+        new URL('/ghost/api/admin/identities/', window.location.origin),
+        handle
+    );
+    return useQuery({
+        queryKey: [`followers:${handle}`],
+        async queryFn() {
+            const followerUrls = await api.getFollowers();
+            const followerActors = await Promise.all(followerUrls.map(url => api.getActor(url)));
```

Comment:
This is going to create a fuck load of API queries - especially for sites with a lot a followers - this looks to me like our API isn't serving you well, and we need to fix it on the backend!

Reply:
@allouis TBH this one is a bit over my head 😅 But looking at [viewFollowersModal.tsx](https://github.com/TryGhost/Ghost/blob/main/apps/admin-x-activitypub/src/components/profile/ViewFollowersModal.tsx) on main, I think that’s how it functioned from the start. But at some point it started rendering an empty list in the frontend, so I thought it’d be just a quick fix so gave it a go with Cursor’s help, should have checked with you guys first 🐒  

So yeah, if it makes more sense to change the API let’s do it! One thing though: I don’t wanna create extra work for you this week since this is not Likes-related, so does it make sense to merge this part as is, and then change that later? Or just revert that commit and revisit it once you change the API?

Reply:
I think revert that commit, and we will live with it being broken until it's fixed at the API layer ☺️